### DISCUSSION TITLE
Setting up the directory structure for a build

### DISCUSSION THREAD

**Discussion by Excelsior2021** [community/community#39667] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have a set-up a repo to use GitHub Pages. It builds fine, however, the directory structure is not quite right.

On my local system the directory structure is something like:

**root**
----index.html
-----**pages**
--------page1.html
--------page2.html

When I access the site at **username.github.io/repo-name**, it loads the index.html fine. But when try the access one of the other pages. For example page1.html it links to **username.github.io/pages/page1.html** instead of **username.github.io/repo-name/pages/page1.html**.

**username.github.io/pages/page1.html** takes me to a non-existent page relative to my user site.
**username.github.io/repo-name/pages/page1.html** takes me to the correct page.

Is there a way I can configure the build so that any files in subdirectories are accessed with the repo-name as the root folder?

**Comment by david-engelmann** [comment, marked as answer]
[This blog post has a great run down on configuring Github pages](https://blog.shahednasser.com/deploy-a-website-with-jekyll-and-github-pages/#deploying-to-github-pages). I'll include the most relevant section below.

```
In _config.yml you need to change three key values:

baseurl: this needs to be the subpath of the website
url: this is the main URL of the website.
destination: The directory in which the build goes to
When creating a website in GitHub pages, usually the format of the URL will be: https://USERNAME.github.io/REPOSITORY-NAME

So, baseurl should be /REPOSITORY-NAME part of your URL, and url should be https://USERNAME.github.io part of your URL.
```

**Reply by Excelsior2021** [reply]
Thanks for the response. I have restructured my directory so that I do not need to configure those settings.

**Reply by david-engelmann** [reply]
Perfect, make sure to mark it as closed!

**Reply by Excelsior2021** [reply]
Not sure how to mark as closed.

**Reply by david-engelmann** [reply]
Clcik "Mark as Answer" button

**Reply by fran1data** [reply]
Hi, how did you restructure it? I am having a the same issue with my page. 
I created a dir's. with 3 sub-dir's., where my main dir. is index.html. The index.html loads and has the 3 links but when I click any of the links I get transferred to a webpage with a 404 error.

**Comment by yoannchaudet** [comment]
There are two kind of Pages sites: https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#types-of-github-pages-sites.

If you want your content to be exposed directly on your username's dedicated hostname (`<user>.github.io`), simply rename your repository to `<user>.github.io`.

**Reply by Excelsior2021** [reply]
Thanks for the response. I already have a user site. I was trying to build a project site. The project site had HTML files in a subdirectory relative to the root.
